File: cpp/Clothing.cpp
// import library
#include <iostream>
#include <string>

// using std namespace
using namespace std;

#include "Product.cpp"
class Clothing : public Product
{
private:
    string size;
    string material;
    string gender;

public:
    Clothing()
    {
    }

    Clothing(string size, string gender, string material)
    {
        this->size = size;
        this->gender = gender;
        this->material = material;
    }

    string getSize()
    {
        return this->size;
    }

    void setSize(string size)
    {
        this->size = size;
    }

    string getMaterial()
    {
        return this->material;
    }

    void setMaterial(string material)
    {
        this->material = material;
    }

    string getGender()
    {
        return this->gender;
    }

    void setGender(string gender)
    {
        this->gender = gender;
    }
    ~Clothing(){};
};

File: php/Clothing.php
<?php
require_once "Product.php";

class Clothing extends Product
{
    private $size;
    private $material;
    private $gender;

    public function __construct()
    {
    }

    // setter getter
    public function  getSize(): string
    {
        return $this->size;
    }

    public function  setSize($size)
    {
        $this->size = $size;
    }

    public function  getMaterial()
    {
        return $this->material;
    }

    public function  setMaterial($material)
    {
        $this->material = $material;
    }

    public function  getGender()
    {
        return $this->gender;
    }

    public function  setGender($gender)
    {
        $this->gender = $gender;
    }
}
